### DISCUSSION TITLE
".zimignore"/.nomedia equivalent for the indexer

### DISCUSSION THREAD

**Discussion by introt** [zim-desktop-wiki/zim-desktop-wiki#2018] [Ideas]
**Is your feature request related to a problem? Please describe.**
I use Zim to document a directory hierarchy. Some of these directories contain a whole lot of sub-directories and files, and despite none of them having any resemblance to a text file, Zim wants to recurse into and index each and every one.

**Describe the solution you'd like**
`exclude_directories=` setting (with globs) under `notebook.zim` > `[Notebook]`, or not recursing into directories that contain a file called `.nomedia` or similar.

**Describe alternatives you've considered**
Not recursing into non-attachment-dir directories doesn't work because non-existing pages can have existing sub pages (in which case the parents lack a .txt, but do exist in the index database), but we could certainly skip indexing files inside such directories as they aren't exposed to the user in any way anyway.

**Additional context**
This function is called for every single file including attachments, ie. in the magnitude of 10e5 times when indexing my _regular_ wiki. Imo the logger statement doesn't belong here in the hot path - `file` is known by the caller anyway.

https://github.com/zim-desktop-wiki/zim-desktop-wiki/blob/89337ddfd8fc7a48906583abca1270e770151186/zim/notebook/index/files.py#L258-L264

Related issues: #306, #907

**Comment by lucvtzel** [comment]
I think the idea is good. In my situation, **indexing deals with directories that should not be indexed**, and indexing lasts a long time, consuming 100% of CPU activity unnecessarily. For instance, with extra library directories contained in attachment folders. Sometimes it even crashes when the indexed file name is invalid.

Other people have also had this type of problem: see https://bugs.launchpad.net/zim/+bug/1199341 .

A `.zimignore` file which would indicate the directories to exclude, could be a solution.
